### PR TITLE
Fix HTTP message formatting to be RFC2616 (section 5.1) compliant, an…

### PR DESCRIPTION
**Pull request by cmacdonald**
Fix HTTP message formatting to be RFC2616 (section 5.1) compliant. 
Hence so they will pass through a transparent Squid proxy.

---

More details:

--------------------------------------------------------------------
Problem The http post syntax.
There should not be a space after HTTP/1.1
--------------------------------------------------------------------
see RFC http://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html
Which says...

"
5.1 Request-Line

The Request-Line begins with a method token, followed by the Request-URI
and the protocol version, and ending with CRLF. The elements are
separated by SP characters. No CR or LF is allowed except in the final
CRLF sequence.

         Request-Line   = Method SP Request-URI SP HTTP-Version CRLF
"

This is also a more readable summary
http://my.safaribooksonline.com/book/networking/security/0201761769/http-1dot1-and-http-1dot0-method-and-field-definitions/app02

In short, remove the space and really the \n should be \r\n

### CODE REVIEW

**Comment by cmacdonald** [comment]
Pinging on this...?

**Comment by pral2a** [comment]
Thanks @cmacdonald!

We are really happy when we receive pull request like yours!
